src/Core/Managed/Shared/DataContracts/AggregatedMetricTelemetry.cs

```diff
@@ -0,0 +1,157 @@
+﻿namespace Microsoft.ApplicationInsights.DataContracts
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.External;
+
+    /// <summary>
+    /// Telemetry type used to track aggregated metric information.
+    /// </summary>
+    public sealed class AggregatedMetricTelemetry : ITelemetry, ISupportProperties
+    {
+        internal const string TelemetryName = "Metric";
+
+        internal readonly string BaseType = typeof(MetricData).Name;
+
+        internal readonly MetricData Data;
+        internal readonly DataPoint Metric;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AggregatedMetricTelemetry"/> class with empty 
+        /// properties.
+        /// </summary>
+        public AggregatedMetricTelemetry()
+        {
+            this.Data = new MetricData();
+            this.Metric = new DataPoint();
+            this.Context = new TelemetryContext(this.Data.properties);
+
+            // We always have a single 'metric'.
+            this.Data.metrics.Add(this.Metric);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AggregatedMetricTelemetry"/> class with properties provided.
+        /// </summary>
+        /// <remarks>
+        /// Metric statistics provided are assumed to be calculated over a period of time equaling 1 minute.
+        /// </remarks>
+        /// <param name="metricName">Metric name.</param>
+        /// <param name="count">Count of values taken during aggregation interval.</param>
+        /// <param name="sum">Sum of values taken during aggregation interval.</param>
+        /// <param name="min">Minimum value taken during aggregation interval.</param>
+        /// <param name="max">Maximum of values taken during aggregation interval.</param>
+        /// <param name="standardDeviation">Standard deviation of values taken during aggregation interval.</param>
+        public AggregatedMetricTelemetry(
```

Comment:
The ctor with parameters exists for the case when people aggregated entirely outside our Sdk and just want to send the results to AI. It states explicitly what is expected. If you use parameter-less ctor and property setters, it is not entirely clear what to set to get consistent results. This is somewhat against the purity of the design, but this is where we hit cold reality. We need to capture most common cases of providing metric data and make sure we do not create a large set of concepts. I thought having two concepts combined into one: a) metric that is just count (sum and count) and I do not care about the rest of the properties and b) metric where all properties are interested; would be Ok.

Therefore, people using parameter-less ctor may choose properties they want to set based on what they are going to look at. People using the other ctor know all things that need to be set if you want to go "comprehensive".